protocol.md

```diff
@@ -397,14 +395,22 @@ This extension defines how to upload finite streams of data that have an
 unknown length at the beginning of the upload.
 
 If the [Creation](#creation) extension is used to initiate a new upload the
-`Upload-Length` header MUST be set to the string `streaming`. Once the total size of the
-entire upload is known it MUST be included as the `Upload-Length` header's value
-in the next `PATCH` request. Once the upload's length has been set it MUST NOT
-be changed.
+`Upload-Stream` header MUST be set to 1. Once the total size of the entire 
+upload is known `Upload-Length` MUST be set in the next `PATCH` request. 
 
 In order to indicate that this extension is supported by the Server it MUST
 include the `stream` element in the `Tus-Extension` header.
 
+#### Headers
```

Comment:
The Headers section contains definitions of all new headers defined by this extension. Please remove `Upload-Length` from this section and move the note about the immutability of the upload's length somewhere else. In addition please add a sentence to `Upload-Stream` that it's usage to indicate the usage of the Stream extension.